### ISSUE TITLE
[BUG] liveMatchId.ToString() is always -1

### ISSUE THREAD

**Issue by borzaka** [shobhit-pathak/MatchZy#255]
I have a multi instance CS2 server: from the same cs2 folder, I run multiple game servers with different ports. I don't know if this is related, or not.

The problem is, when I set up matches width `matchzy_loadmatch`, the force name file is always `Match_-1.ini` in the `MatchZyPlayerNames` folder.
So, it keeps overwriting, and contains just the last loaded match player names. And later, players with differnt names can join the server.

**Comment by shobhit-pathak** [comment]
Hey!

Are you passing matchid in your JSON file while loading the matches? And do you see any error in console (probably DB related) on loading the match?

**Comment by borzaka** [comment]
No, I didn't set `matchid` in the match config. I except that it auto increments based on database. Every other place, the match id is auto incremented;
- in the database
- in the demo filename
- in the backup filename

Except in the `MatchZyPlayerNames` folder, which causes the above problem where not every player name got forced.

**Comment by borzaka** [comment]
Stats and data backup has correct auto increment ID:
![image](https://github.com/user-attachments/assets/8be22ddf-cd41-4a4e-a746-0ffacc2899c4)

But not for MatchZyPlayerNames:
![image](https://github.com/user-attachments/assets/5876b308-ab02-49f2-b734-fe1bdb031d83)

**Comment by borzaka** [comment]
My proposal:
If `matchid` is missing from the match JSON, the plugin will automatically use an incremented ID instead. As a result, the `Match_X.ini` file will contain the correct ID and load it correctly.

So, this is no true for the `Match_X.ini` file, only for everything else:
![Image](https://github.com/user-attachments/assets/d75ae8dd-6c2b-4f52-94c3-765e7c7e85d3)
